packages/rollup-plugin-html/test/rollup-plugin-html.test.ts

```diff
@@ -846,4 +846,56 @@ describe('rollup-plugin-html', () => {
       '<html><head></head><body><img src="./image-c.png"><link rel="stylesheet" href="./styles.css"><img src="./image-b.svg"></body></html>',
     );
   });
+
+  it.only('will inject service worker registration code if a service worker path is defined by another plugin', async () => {
+    function createServiceWorker() {
+      return {
+        name: 'create-service-worker',
+        // @ts-ignore
+        async resolveId(source, importer) {
```

Comment:
Can you explain what's going on in this function? Shouldn't you check the actual import being resolved is the service worker's? It seems like this would match all entrypoints?